java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -1321,8 +1321,14 @@ public final Builder append(BigDecimal value, RoundingMode roundingMode) {
       assert currentIndex < rows;
       BigInteger unscaledValue = value.setScale(-type.getScale(), roundingMode).unscaledValue();
       if (type.typeId == DType.DTypeEnum.DECIMAL32) {
+        if (value.precision() > DType.DECIMAL32_MAX_PRECISION) {
```

Comment:
I am not sure if we need this check and throw Exception here. Or should we just remove this and let it throw from later